data/plugins/projectsearch.lua

```diff
@@ -30,8 +30,14 @@ local function find_all_matches_in_file(t, filename, fn)
   for line in fp:lines() do
     local s = fn(line)
     if s then
-      table.insert(t, { file = filename, text = line, line = n, col = s })
-      core.redraw = true
+      -- Insert maximum 256 characters. If we insert more, for compiled files, which can have very long lines
+      -- things tend to get sluggish. If our line is longer than 80 characters, begin to truncate the thing.
+      if #line > 80 then
+        table.insert(t, { file = filename, text = (s > 80 and "..." or "") .. line:sub(math.max(s - 80, 1), s + (256 - 80)), line = n, col = s })
```

Comment:
It is not lear to me if by truncating we are losing the ability to find a text in the line.

Reply:
Shouldn't; we're calling sub on line, so that we always display the relevant bit of the line (starting from 80 characters earlier, prepended with an ellipsis). I could tweak this to be relevant to the size of the screen, if you like.

As it stands, the current implementation doesn't allow for finding the text you're searching for, because you can't horizontally scroll the results pane anyway. This actually shows the relevant text immediately in the main window.